flyteadmin/pkg/manager/impl/task_manager.go

```diff
@@ -89,14 +89,16 @@ func (t *TaskManager) CreateTask(
 		return nil, err
 	}
 	// See if a task exists and confirm whether it's an identical task or one that with a separate definition.
-	existingTask, err := util.GetTaskModel(ctx, t.db, request.Spec.Template.Id)
+	existingTaskModel, err := util.GetTaskModel(ctx, t.db, request.Spec.Template.Id)
 	if err == nil {
-		if bytes.Equal(taskDigest, existingTask.Digest) {
-			return nil, errors.NewFlyteAdminErrorf(codes.AlreadyExists,
-				"identical task already exists with id %s", request.Id)
+		if bytes.Equal(taskDigest, existingTaskModel.Digest) {
+			return nil, errors.NewTaskExistsIdenticalStructureError(ctx, &request)
 		}
-		return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument,
-			"task with different structure already exists with id %v", request.Id)
+		existingTask, err2 := util.GetTask(ctx, t.db, *request.Id)
+		if err2 != nil {
```

Comment:
we already have `existingTaskModel` fetched above, can we just call https://github.com/flyteorg/flyte/blob/master/flyteadmin/pkg/repositories/transformers/task.go#L41 here instead?

Reply:
Great! Thanks for the heads-up.